gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/service/MinioHandler.java

```diff
@@ -120,26 +131,27 @@ public void updateTasks(Event event) {
                             addWaitingFileAndNotifyTasks(processFileMinio, runningOrPendingTasks);
                         } else {
                             processFile = processFileRepository.save(processFile);
-                            Set<TaskWithStatusUpdate> taskWithStatusUpdates = taskService.addProcessFileToTasks(processFile, processFileMinio.getFileEventType(), true, true);
+                            Set<TaskWithStatusUpdate> taskWithStatusUpdates = taskService.addProcessFileToTasks(processFile,
+                                                                                                                processFileMinio.getFileEventType(), true, true);
                             saveAndNotifyTasks(taskWithStatusUpdates, true);
                             LOGGER.info("Process file {} has been added properly", processFile.getFilename());
                         }
                     }
                 } else {
-                    String objectKey = URLDecoder.decode(event.objectName(), StandardCharsets.UTF_8);
-                    LOGGER.warn("Minio object {} has not been added ", objectKey);
+                    LOGGER.warn("Minio object {} has not been added ", URLDecoder.decode(event.objectName(), UTF_8));
                 }
             }
         }
     }
 
     private ProcessFileMinio buildProcessFileMinioFromEvent(Event event) {
-        String validityInterval = event.userMetadata().get(FILE_VALIDITY_INTERVAL_METADATA_KEY);
-        String objectKey = URLDecoder.decode(event.objectName(), StandardCharsets.UTF_8);
-        if (validityInterval != null && !validityInterval.isEmpty()) {
-            String fileGroup = event.userMetadata().get(FILE_GROUP_METADATA_KEY);
-            String fileType = event.userMetadata().get(FILE_TYPE_METADATA_KEY);
-            String documentId = event.userMetadata().get(DOCUMENT_ID_METADATA_KEY);
+        final Map<String, String> metadata         = event.userMetadata();
+        final String              validityInterval = metadata.get(FILE_VALIDITY_INTERVAL_METADATA_KEY);
+        final String objectKey = URLDecoder.decode(event.objectName(), UTF_8);
+        if (!isEmpty(validityInterval))  {
+            String fileGroup = metadata.get(FILE_GROUP_METADATA_KEY);
```

Comment:
I'm surprised you didn't mark the Strings as final here. :stuck_out_tongue_closed_eyes: